kore/src/Kore/Unification/UnifierT.hs

```diff
@@ -51,7 +51,20 @@ newtype UnifierT (m :: Type -> Type) a =
                 (LogicT m)
                 a
         }
-    deriving newtype (Functor, Applicative, Monad, Alternative, MonadPlus)
+    deriving newtype (Functor, Applicative)
+
+instance Alternative (UnifierT m) where
+    empty = UnifierT $ lift empty
+    x <|> y = UnifierT $ getUnifierT x <|> getUnifierT y
+
+instance Monad (UnifierT m) where
+    UnifierT {getUnifierT = unifier} >>= cont =
+        UnifierT $ unifier >>= getUnifierT . cont
+    return = pure
+
+instance MonadPlus (UnifierT m) where
+    mzero = empty
+    mplus = (<|>)
```

Comment:
Are these different than the derived instances?

Reply:
I don't know exactly what the derived instances would look like but I expect they would look a lot like this

Reply:
Why did we add these hand-written instances, then?